source/agora/cli/client/GenTxProcess.d

```diff
@@ -0,0 +1,176 @@
+import agora.utils.PrettyPrinter;
+import agora.utils.Test;
+
+import core.time;
+import std.algorithm;
+import std.format;
+import std.getopt;
+import std.range;
+import std.stdio;
+
+import vibe.core.core;
+
+public alias APIMaker = API delegate (string address);
+
+/// Option required to generate and send transactions
+private struct GenTxOption
+{
+    /// IP address of node
+    public string host = "0.0.0.0";
+
+    /// Port of node
+    public ushort port = 2826;
+
+    /// The address key to send the output
+    public string address;
+
+    /// Interval of sending transaction
+    public uint interval = 5;
+
+    /// dump output option
+    public bool dump;
+}
+
+/// Parse the command-line arguments of gentx
+public GetoptResult parseGenTxOption (ref GenTxOption op, string[] args)
+{
+    return getopt(
+        args,
+        "ip|i",
+            "IP address of node (default: 0.0.0.0)",
+            &op.host,
+
+        "port|p",
+            "Port of node (default: 2826)",
+            &op.port,
+
+        "dest|d",
+            "The address key to send the output",
+            &op.address,
+
+        "interval|t",
+            "Interval of sending transaction",
+            &op.interval,
+
+        "dump|o",
+            "dump output option",
+            &op.dump
+            );
+}
+
+/// Print help
+public void printGenTxHelp (ref string[] outputs)
+{
+    outputs ~= "usage: agora-client gentx [--dump] [--ip] [--port] [--dest] [--interval]";
+    outputs ~= "";
+    outputs ~= "   gentx      Generate and send a transaction to node";
+    outputs ~= "";
+    outputs ~= "        -i --ip         IP address of node";
+    outputs ~= "        -p --port       Port of node";
+    outputs ~= "        -d --dest       The address key to send the output";
+    outputs ~= "        -t --interval   Interval of sending transaction";
+    outputs ~= "        -o --dump       Dump output option";
+    outputs ~= "";
+}
+
+/*******************************************************************************
+
+    Input arguments, generate the transaction and send it to the node
+
+    Params:
+        args = client command line arguments
+
+*******************************************************************************/
+
+public int genTxProcess (string[] args, ref string[] outputs,
+                           APIMaker api_maker)
+{
+    GenTxOption op;
+    GetoptResult res;
+
+    try
+    {
+        res = parseGenTxOption(op, args);
+        if (res.helpWanted)
+        {
+            printGenTxHelp(outputs);
+            return CLIENT_SUCCESS;
+        }
+    }
+    catch (Exception ex)
+    {
+        printGenTxHelp(outputs);
+        return CLIENT_EXCEPTION;
+    }
+
+    bool isValid = true;
+
+    if (op.address.length == 0)
+    {
+        if (isValid) printGenTxHelp(outputs);
+        outputs ~= "Address is not entered. [--dest]";
+        isValid = false;
+    }
+
+    if (!isValid)
+        return CLIENT_INVALID_ARGUMENTS;
```

Comment:
In general we should always prefer setting a validity flag to false by default. It's good programming style because if you forget to set it to false then the rest of the code assumes the flag is correct. It's better to assume it's invalid by default.

But in any case you don't really need a flag here.

```suggestion
    if (op.address.length == 0)
    {
        printGenTxHelp(outputs);
        outputs ~= "Address is not entered. [--dest]";
        return CLIENT_INVALID_ARGUMENTS;
    }
```